### PR TITLE
feat: enable csv and excel downloads to be filtered

### PR DESCRIPTION
**Pull request by mswertz**
motivation: in vac4eu the will want to download filtered csv from their TableMappings view (todo in next PR).

current PR: enable filtered excel and csv download in table explorer.

to test, in table explorer apply some filter, then choose the 'download' button, and select 'filtered' csv and excel.

Also, 
- I removed the 'includeSystemColumns' logic to outside the IO package.
- and removed the magic 'if you don't provide columNames we will download all columns'; now you have to provide the columns you want.

Possible todo: also enable on csv.zip

### CODE REVIEW

**Comment by mswertz** [comment]
[#9611 VAC4EU [part 2]: can download CSV and Excel files, filtered](https://trello.com/c/14q2t1uW/608-9611-vac4eu-part-2-can-download-csv-and-excel-files-filtered)

File: apps/molgenis-components/src/components/tables/TableExplorer.vue [hunk 1]
```diff
@@ -34,18 +34,64 @@
...
+                  |
+                  <ButtonAlt
+                    :href="
+                      '/' +
+                      schemaName +
+                      '/api/excel/' +
+                      tableId +
+                      '?filter=' +
+                      JSON.stringify(graphqlFilter)
+                    "
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/tables/TableExplorer.vue:78]: I really do not think the current code format is helpful here, de vue default does a better job

**Review by connoratrug — APPROVED** [review]
nice refactor.

Maybe the ui could improve by flipping the link with the type ( all / filtered csv , instead of csv all || filtered ) , or using two columns and left aligning the download type. 

Maybe also just show the type ( csv, exel, ..) as link when so filter is active ( 'all rows' could be confusing in the context)

File: apps/molgenis-components/src/components/tables/TableExplorer.vue (hunk 1 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/tables/TableExplorer.vue:78]: I do agree, we should see if we can change this formatting somehow to be more reasonable.

**Pull request merged**